daemon/graphdriver/overlay2/overlay.go

```diff
@@ -191,11 +192,11 @@ func Init(home string, options []string, uidMaps, gidMaps []idtools.IDMap) (grap
 		// Try to enable project quota support over xfs.
 		if d.quotaCtl, err = quota.NewControl(home); err == nil {
 			projectQuotaSupported = true
+		} else if opts.quota.Size > 0 {
+			return nil, fmt.Errorf("Project quota is not supported: %v", err)
```

Comment:
As @rhvgoyal noted correctly, this test should move outside the backingFs == "xfs" block
(i.e. if  !projectQuotaSupported) and error message should be made more clear to the user, something like the error in Create():
"overlay2.size option is supported only for overlay over xfs with 'pquota' mount option"